orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.cc

```diff
@@ -428,30 +463,56 @@ Status PaddingElimination::ApplyImpl(Graph& graph, bool& modified, int graph_lev
   NodeArg* squeeze_out_arg = InsertNodesForValidIndices(
       graph, reshape_output_args[0], embedding_node->MutableInputDefs()[2], embedding_node->GetExecutionProviderType());
 
+  std::vector<int64_t> first_two_indices{0, 1};
+  ONNX_NAMESPACE::TensorProto first_two_indices_const_tensor;
+  first_two_indices_const_tensor.set_name(graph.GenerateNodeArgName("first_two_indices"));
+  first_two_indices_const_tensor.set_data_type(ONNX_NAMESPACE::TensorProto_DataType_INT64);
+  first_two_indices_const_tensor.add_dims(first_two_indices.size());
+  first_two_indices_const_tensor.set_raw_data(first_two_indices.data(), first_two_indices.size() * sizeof(int64_t));
+  NodeArg* first_two_indices_arg = &graph_utils::AddInitializer(graph, first_two_indices_const_tensor);
+  // Get the first two dims value of input_ids which is [batch_size, seq_len]
+  NodeArg* first_two_dims_arg = GetDimsValue(graph, input_ids_arg, first_two_indices_arg, *embedding_node);
```

Comment:
shall we move this into InsertExpandForNodeInput, looks first_two_dims_arg  is needed only by then.

Reply:
No, it's also needed in InsertNodesForOutput.